recipes/gtsam/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+class gtsamConan(ConanFile):
+    name = "gtsam"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/borglab/gtsam"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("GTSAM is a library of C++ classes that implement\
+                    smoothing and mapping (SAM) in robotics and vision")
+    topics = ("conan", "gtsam", "mapping", "smoothing")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_quaternions": [True, False],
+               "pose3_expmap": [True, False],
+               "rot3_expmap": [True, False],
+               "enable_consistency_checks": [True, False],
+               "with_TBB": [True, False],
+               "with_eigen_MKL": [True, False],
+               "with_eigen_MKL_OPENMP": [True, False],
+               "throw_cheirality_exception": [True, False],
+               "allow_deprecated_since_V4": [True, False],
+               "typedef_points_to_vectors": [True, False],
+               "support_nested_dissection": [True, False],
+               "tangent_preintegration": [True, False],
+               "build_wrap": [True, False],
+               "wrap_serialization": [True, False],
+               "build_unstable": [True, False],
+               "disable_new_timers": [True, False],
+               "build_type_postfixes": [True, False],
+               "install_matlab_toolbox": [True, False],
+               "install_cython_toolbox": [True, False],
+               "install_cppunitlite": [True, False],
+               "install_geographiclib": [True, False]}
+
+    default_options = {"shared": False,
+                       "fPIC": True,
+                        "use_quaternions": False,
+                        "pose3_expmap": False,
+                        "rot3_expmap": False,
+                        "enable_consistency_checks": False,
+                        "with_TBB": False,
+                        "with_eigen_MKL": False,
+                        "with_eigen_MKL_OPENMP": False,
+                        "throw_cheirality_exception": True,
+                        "allow_deprecated_since_V4": True,
+                        "typedef_points_to_vectors": False,
+                        "support_nested_dissection": False,
+                        "tangent_preintegration": False,
+                        "build_wrap": False,
+                        "wrap_serialization": True,
+                        "build_unstable": True,
+                        "disable_new_timers": False,
+                        "build_type_postfixes": True,
+                        "install_matlab_toolbox": False,
+                        "install_cython_toolbox": False,
+                        "install_cppunitlite": True,
+                        "install_geographiclib": False}
```

Comment:
Geographiclib could be external. Bincrafters has a recipe for it: https://github.com/bincrafters/conan-geographiclib, we can port it to CCI too.

Reply:
Working on it.

Reply:
It's merged now, but the flag GTSAM_INSTALL_GEOGRAPHICLIB only compiles and installs the library. GTSAM doesn't use it explicitly so I think it wouldn't be appropriate to add it as a requirement.
So we could either force this flag to false and if the consumer wants geographiclib he/she should add it as a dependency for the project or we could leave it as an option just in case the consumer wants to build from this sources.
For my specific use case I don't need geographiclib so I don't mind.

Reply:
ping @uilianries :)

Reply:
@danimtb I completely forgot about this recipe tbh. This is where it got stuck if I recall correctly.

Reply:
@ericriff I agree, lets force to False and add it as dependency.